### DISCUSSION TITLE
GitHub Branch Policies?

### DISCUSSION THREAD

**Discussion by ryhsh72** [community/community#21921] [New to GitHub]
<p>Hi everyone!</p>
<p>I’m trying to see if GitHub has a feature similar to VSTS (Microsoft) where you can institute <strong>branch policies</strong> , or requirements that prevent direct pushing of commits to a particular branch. In this case, users would have to make a pull-request in order to push code into its upstream.</p>
<p>Is there any option like that?</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/ryhsh72">@ryhsh72</a>,</p>
<p>Yes, GitHub does have something like that! You can implement this sort or restriction through <a href="https://help.github.com/articles/about-protected-branches/" rel="nofollow noopener">GitHub’s protected branches</a>. You can even <a href="https://help.github.com/articles/approving-a-pull-request-with-required-reviews/" rel="nofollow noopener">require that all pull requests require reviews</a> before they get merged into the protected branch.</p>
<p>Does that help?</p>
<p>Thanks!</p>

**Comment by ryhsh72** [comment]
<p>That is exactly what I needed, thank you <a class="mention" href="/that-pat">@that-pat</a>! :smiley:</p>

**Comment by ThirdEyeRose** [comment]
<p>You’re very welcome!</p>
